components/DiscussionThread/DiscussionThread.tsx

```diff
@@ -0,0 +1,109 @@
+import React, { useState } from "react";
+import ReactMarkdown from "react-markdown";
+import TimeAgo from "react-timeago";
+import { AnchorButton, Button, FormGroup, Intent, TextArea } from "@blueprintjs/core";
+
+import styles from "./DiscussionThread.module.scss";
+import Avatar from "components/Avatar/Avatar";
+import { useLoginContext } from "utils/client/hooks";
+import classNames from "classnames";
+
+interface Props {
+	initDiscussionThread: any;
+	minimal?: boolean;
+}
+
+const DiscussionThread: React.FC<Props> = function ({ initDiscussionThread, minimal }) {
+	const loginData = useLoginContext();
+
+	const [discussionThread, setDiscussionThread] = useState(initDiscussionThread);
+
+	const [replyText, setReplyText] = useState("");
+	const [isLoading, setIsLoading] = useState(false);
+	const handleSubmit = async () => {
+		setIsLoading(true);
+		const response = await fetch("/api/discussionItem", {
+			method: "POST",
+			headers: { "Content-Type": "application/json" },
+			body: JSON.stringify({
+				text: replyText,
+				discussionThreadId: discussionThread.id,
+			}),
+		});
+		const newDiscussionItem = await response.json();
+		setDiscussionThread({
+			...discussionThread,
+			discussionItems: [...discussionThread.discussionItems, newDiscussionItem],
+		});
+		setReplyText("");
+		setIsLoading(false);
+	};
+	return (
+		<div className={minimal ? styles.minimal : ""}>
+			<h2 className={styles.threadTitle}>
+				{discussionThread.title}{" "}
+				<span className={styles.number}>#{discussionThread.number}</span>
+				{minimal && (
+					<AnchorButton
+						className={styles.permalinkButton}
+						minimal
+						small
+						href={`discussions/${discussionThread.number}`}
+						icon="link"
+					/>
+				)}
+			</h2>
+			{discussionThread?.discussionItems.map((item: any) => {
```

Comment:
Running into `TypeError: Cannot read properties of undefined (reading 'map')` here. Looked a bit into this and for newly created `discussionThread`, `discussionThread.discussionItems` can be undefined.

Reply:
Nice catch, thanks! Fixed.